crypto.go

```diff
@@ -36,7 +28,7 @@ func _KeyFromProtobuf(pbKey *services.Key) (Key, error) {
 	}
 	switch key := pbKey.GetKey().(type) {
 	case *services.Key_Ed25519:
-		return PublicKeyFromBytes(key.Ed25519)
+		return Ed25519PublicKeyFromBytes(key.Ed25519)
```

Comment:
We should have such a method as public. Instead we should have `PublicKeyFromBytesEd25519()`. The reason this distinction is necessary is because `Ed25519*` should not be public as it's a private type, but `PublicKey` is public and hence should have public methods for constructing keys from `Ed25519`.